src/Symfony/Component/Notifier/Message/WebPushMessage.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Message;
+
+use Symfony\Component\Notifier\Notification\Notification;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class WebPushMessage implements MessageInterface
+{
+    private $transport;
+    private $subject;
+    private $content;
+    private $options;
+    private $notification;
+
+    public function __construct(string $subject, string $content, MessageOptionsInterface $options = null)
+    {
+        $this->subject = $subject;
+        $this->content = $content;
+        $this->options = $options;
+    }
+
+    public static function fromNotification(Notification $notification): self
+    {
+        $message = new self($notification->getSubject(), $notification->getContent());
+        $message->notification = $notification;
+
+        return $message;
+    }
+
+    /**
+     * @return $this
```

Comment:
This comment is not needed. You see that from the return type. Same with the others.